gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -54,6 +77,81 @@ public class GoogleCloudStorageClientImpl extends ForwardingGoogleCloudStorage {
             .setHttpRequestInitializer(httpRequestInitializer)
             .setDownscopedAccessTokenFn(downscopedAccessTokenFn)
             .build());
+    this.storageOptions = options;
+    this.storage =
+        clientLibraryStorage == null ? createStorage(credentials, options) : clientLibraryStorage;
+  }
+
+  @Override
+  public WritableByteChannel create(StorageResourceId resourceId, CreateObjectOptions options)
+      throws IOException {
+    logger.atFiner().log("create(%s)", resourceId);
+    checkArgument(
+        resourceId.isStorageObject(), "Expected full StorageObject id, got %s", resourceId);
+    // Update resourceId if generationId is missing
+    StorageResourceId resourceIdWithGeneration = resourceId;
+    if (!resourceId.hasGenerationId()) {
+      resourceIdWithGeneration =
+          new StorageResourceId(
+              resourceId.getBucketName(),
+              resourceId.getObjectName(),
+              getWriteGeneration(resourceId, options.isOverwriteExisting()));
+    }
+
+    GoogleCloudStorageClientWriteChannel channel =
+        new GoogleCloudStorageClientWriteChannel(
+            storage, storageOptions, resourceIdWithGeneration, options, backgroundTasksThreadPool);
+    channel.initialize();
+    return channel;
+  }
+
+  @Override
+  public void close() {
+    try {
+      try {
+        super.close();
+      } finally {
+        backgroundTasksThreadPool.shutdown();
+      }
+    } catch (Exception e) {
+      logger.atWarning().withCause(e).log("Exception while closing gcsClientImpl");
```

Comment:
why are we ignoring this exception? If this fails then we might not have written all the data, correct? Hence this can result in data loss, correct?

Reply:
No, this is not WriteChannel `close` this is close of a `GoogleCloudStorage` which does nothing apart from cleaning up the background thread pool. Also it shutdowns the `stubProvider` `watchDog`.

We can ignore the exception here because any functionality is not getting here.

Reply:
What is the issue if we rethrow? Atleast caller has a chance to act on it. That seem to be a safer option.

Reply:
Removing the try-catch and let called decide what needs to be done. instead of absorbing exception.